experimental/native_file_system/virtual_root_provider_win.cc

```diff
@@ -1,12 +1,63 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
+
+#include <shlobj.h>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+
+bool VirtualRootProvider::testing_enabled_ = false;
+
+VirtualRootProvider::VirtualRootProvider() {
+  if (!virtual_root_map_.empty())
+    return;
+
+  if (testing_enabled_)
+    base::GetTempDir(&home_path_);
+  else
+    home_path_ = base::GetHomeDir();
```

Comment:
`home_path_` is useless in Windows. So, please leave it empty. 
All SHGetKnownFolderPath() initialized paths will be moved into this `else` branch.